File: product-processor-service/src/main/java/com/example/product_processor_service/configuration/kafka/topic/initializer/KafkaTopicInitializer.java
package com.example.product_processor_service.configuration.kafka.topic.initializer;

import com.example.product_processor_service.configuration.kafka.environment.KafkaEnvironment;
import org.apache.kafka.clients.admin.NewTopic;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.kafka.config.TopicBuilder;

@Configuration
public class KafkaTopicInitializer {
    private final KafkaEnvironment kafkaEnvironment;

    public KafkaTopicInitializer(KafkaEnvironment kafkaEnvironment) {
        this.kafkaEnvironment = kafkaEnvironment;
    }

    @Bean
    public NewTopic updateProductTopicInitializer() {
        return TopicBuilder
                .name(kafkaEnvironment.getKAFKA_TOPIC_PRODUCT_UPDATE_NAME())
                .replicas(kafkaEnvironment.getKAFKA_REPLICATION_FACTOR())
                .partitions(kafkaEnvironment.getKAFKA_REPLICATION_FACTOR())
                .build();
    }

    @Bean
    public NewTopic deleteAccountProductTopicInitializer() {
        return TopicBuilder
                .name(kafkaEnvironment.getKAFKA_TOPIC_ACCOUNT_PRODUCT_DELETE_NAME())
                .replicas(kafkaEnvironment.getKAFKA_REPLICATION_FACTOR())
                .partitions(kafkaEnvironment.getKAFKA_REPLICATION_FACTOR())
                .build();
    }
}

File: product-manager-service/src/main/kotlin/com/example/product_manager_service/configuration/kafka/topic/initializer/KafkaTopicInitializer.kt
package com.example.product_manager_service.configuration.kafka.topic.initializer

import com.example.product_manager_service.configuration.kafka.environment.KafkaEnvironment
import org.apache.kafka.clients.admin.NewTopic
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.kafka.config.TopicBuilder

@Configuration
class KafkaTopicInitializer(private val kafkaEnvironment: KafkaEnvironment) {
    @Bean
    fun productSaveTopicInitializer(): NewTopic =
        TopicBuilder
            .name(kafkaEnvironment.KAFKA_TOPIC_PRODUCT_SAVE_NAME)
            .replicas(kafkaEnvironment.KAFKA_REPLICATION_FACTOR)
            .partitions(kafkaEnvironment.KAFKA_REPLICATION_FACTOR)
            .build()
}